plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstanceRuntimeInfo.java

```diff
@@ -355,4 +378,18 @@ public String projectsRoot() {
 
         return serversConf;
     }
+
+    /**
+     * Checks environment variable {@code CHE_DOCKER_USE_INTERNAL_CONTAINER_ADDRESS} to determine if
+     * direct addresses of containers should be used.
+     * @return true if {@code CHE_DOCKER_USE_INTERNAL_CONTAINER_ADDRESS} is "true", false otherwise.
+     */
+    protected boolean useInternalContainerAddresses() {
+        String useInternalContainerAddresses = System.getenv(CHE_DOCKER_USE_INTERNAL_CONTAINER_ADDRESS);
```

Comment:
Thank you, that makes a lot of sense! I've updated the PR to use a property (currently called `che.docker.ip.useinternaladdress` -- open to suggestions).

A few questions regarding the conversion of env vars to properties though: 

1) in `CheBootstrap`,  I see that env vars are converted by lowercasing them and replacing underscores with periods. However, some of the properties have underscores in their name, which makes it impossible to set them with environment variables, and prevents the new property here from having underscores (`che.docker.ip.use_internal_address` would make more sense).

2) In `LocalDockerInstanceRuntimeInfo`, it looks like this convention is ignored: the methods `externalHostnameWithPrecedence` and `internalHostnameWithPrecedence` use environment variables to override their behavior -- is this intentional or should it be fixed?